src/Rules/Classes/ImpossibleInstanceOfRule.php

```diff
@@ -79,20 +81,28 @@ public function processNode(Node $node, Scope $scope): array
 				$addTip(RuleErrorBuilder::message(sprintf(
 					'Instanceof between %s and %s will always evaluate to false.',
 					$expressionType->describe(VerbosityLevel::typeOnly()),
-					$classType->describe(VerbosityLevel::typeOnly()),
+					$this->getClassTypeAsString($classType),
```

Comment:
This is the wrong fix. Instead of `VerbosityLevel::typeOnly()`, you need to call `VerbosityLevel::getRecommendedLevelByType()`.

Reply:
OK, I changed it.

This looked weird to me:
```
Instanceof between string and \'str\' will always evaluate to false.
```
That's why I used VerbosityLevel::precise() only for class strings